frontend/src/redux/reducers/ui.tsx

```diff
@@ -291,6 +303,21 @@ function reducer(state = _.cloneDeep(INITIAL_STATE), action: Action) {
       newFilters.isVersionDialogOpen = action.isVersionDialogOpen;
       break;
     }
+    case UI_SETTINGS_TABLE_ROWS_PER_PAGE_OPTIONS: {
+      newFilters.settings.tableRowsPerPageOptions = action.newSetOfOptions;
+      localStorage.setItem('settings', JSON.stringify(newFilters.settings));
+      break;
+    }
+    case UI_SET_SETTINGS: {
+      newFilters.settings = action.settings;
+      localStorage.setItem('settings', JSON.stringify(newFilters.settings));
+      break;
+    }
+    case UI_SETTINGS_TIMEZONE: {
+      newFilters.settings.timezone = action.newTimezone;
+      localStorage.setItem('settings', JSON.stringify(newFilters.settings));
+      break;
+    }
```

Comment:
Since this is all under the same logical set, maybe we should just have a single setter for this, and we could copy only the settings whose keys are known (declared in the initial state).
This would allow us to extend the settings in the future without a lot of changes to the redux mechanism.

Reply:
updated